thrall/app/lib/ThrallMessageConsumer.scala

```diff
@@ -115,16 +118,131 @@ class ThrallMessageConsumer(
     )
   }
 
+  def updateImagePhotoshoot(message: JsValue) = {
+    (message \ "data").validate[Edits].fold(
+      err => {
+        val msg = s"Unable to parse message as Edits ${JsError.toJson(err).toString}"
+        GridLogger.error(msg)
+        Future.failed(SNSBodyParseError(msg))
+      },
+      upcomingEdits => withImageId(message) { id => {
+        GridLogger.info("updating photoshoot", id)
+
+        es.getImage(id) map {
+          case Some(image) => {
+            image.syndicationRights match {
+              case Some(rights) if !rights.isInferred => notInferred(image, upcomingEdits)
+              case _ => inferred(image, upcomingEdits)
+            }
+
+            updateImageUserMetadata(message)
+          }
+          case _ => {
+            GridLogger.info(s"image not found", id)
+            None
+          }
+        }
+      }}
+    )
+  }
+
+  private def deleteInferredRights(message: JsValue) = {
+    Future.sequence(
+      withImageId(message) { id =>
+        GridLogger.info("deleting inferred rights", id)
+        es.deleteSyndicationRights(id)
+      }
+    )
+  }
+
+  private def notInferred(image: Image, upcomingEdits: Edits) = {
```

Comment:
This is long and has a lots of duplication, maybe we can improve by having smaller reusable steps?